### PR TITLE
Add search place to backend

### PR DESCRIPTION
**Pull request by antonoberg**

### CODE REVIEW

File: packages/backend/src/lib/data/hooks/createIndexHook.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+module.exports = function createIndexHook({
+  createIndexJob,
+  rebuildInProgress,
+  resource,
+}) {
+  return function indexHook({ item, serviceInteractor }) {
+    return Promise.resolve().then(() => {
+      const id = item && item.id
```
Comment (fredrikolovsson) [inline packages/backend/src/lib/data/hooks/createIndexHook.js]: should this hook bail if there's no id? (if item.id is guaranteed by validation to be here then we could skip the `item &&`)

File: packages/backend/src/lib/data/hooks/createConsolidateJobsPostHook.js [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+            objectPath.get(job, 'attributes.operationId') ===
+            rebuildViewOperationId
+          )
+        })
+
+        return Promise.resolve()
+          .then(() => {
+            if (containsRebuildJob) {
+              log.debug(
+                `Contains rebuild job will replace all with rebuild job`
```
Comment (fredrikolovsson) [inline packages/backend/src/lib/data/hooks/createConsolidateJobsPostHook.js]: suggested wording `Jobs to consolidate contain rebuild job. Will replace all jobs with rebuild job.`

File: packages/backend/src/lib/data/hooks/createConsolidateJobsPostHook.js [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+              )
+              return createIndexJob({
+                rebuild: true,
+                serviceInteractor,
+              })
+            }
+            log.debug(
+              `Dont contain rebuild job will replace ${
```
Comment (fredrikolovsson) [inline packages/backend/src/lib/data/hooks/createConsolidateJobsPostHook.js]: `Jobs to consolidate does not contain rebuild job. Will replace ${jobsToConsolidate.length} jobs.`

File: packages/scripts/src/bash/elasticsearch-export-indices.sh [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+while getopts f: option
+ do
+  case "${option}"
+   in
+    f) FILE_PATH=${OPTARG};
+  esac
+ done
+
+$FULL_PATH/elasticsearch-export-index.sh -f $FILE_PATH -n searchspecimen
```
Comment (fredrikolovsson) [inline packages/scripts/src/bash/elasticsearch-export-indices.sh]: neat

File: package.json [hunk 5]
```diff
@@ -60,7 +60,7 @@
     "pretty:ui": "cd ./packages/ui && yarn pretty",
     "preversion": "./packages/scripts/src/bash/release-pre-version.sh",
     "remote:build:data:zip": "node ./packages/scripts/src/js/packageJsonDataFiles.js",
-    "remote:build:index:specimen": "node ./packages/scripts/src/js/exec.js docker-rebuild-search.sh",
+    "remote:build:indices": "node ./packages/scripts/src/js/exec.js docker-rebuild-search.sh",
```
Comment (fredrikolovsson) [inline package.json]: maybe it should be called `remote:build:elastic-indices` for consistency (and to indicate it's not postgres indices)

File: packages/backend/src/lib/models/elasticsearch/utilities/createMappingsFromSpecification.js [hunk 6]
```diff
@@ -4,13 +4,25 @@ module.exports = function createMappingsFromSpecification({
...
     (obj, key) => {
-      const {
-        elasticsearch: elasticsearchFunction,
-        fieldPath,
-      } = mappingSpecification.mappings[key]
+      const keyMappingSpecifications = mappingSpecification.mappings[key]
+
+      const keyMappings = {}
+
+      let keyMappingSpecificationsArray = []
```
[the remark this answers is not in the document]
Comment (antonoberg) [reply packages/backend/src/lib/models/elasticsearch/utilities/createMappingsFromSpecification.js]: fixed

File: packages/backend/src/lib/data/mappings/factories/createStringSearchMapping.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
+module.exports = function createKeywordMapping({ fieldPath, searchFieldPath }) {
```
[the remark this answers is not in the document]
Comment (antonoberg) [reply packages/backend/src/lib/data/mappings/factories/createStringSearchMapping.js]: fixed

File: packages/backend/src/lib/data/filters/factories/createStringRegexpSearchFilter.js [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+          must: [],
+        },
+      }
+
+      const regexpFilters = createRegexpElasticFilters({
+        path: fieldPath,
+        value,
+      })
+
+      baseQuery.bool.must = [...baseQuery.bool.must, ...regexpFilters]
```
[the remark this answers is not in the document]
Comment (antonoberg) [reply packages/backend/src/lib/data/filters/factories/createStringRegexpSearchFilter.js]: fixed

File: packages/backend/src/lib/data/hooks/createIndexHook.js (hunk 1 above)
Comment (antonoberg) [reply packages/backend/src/lib/data/hooks/createIndexHook.js]: fixed

File: packages/backend/src/lib/data/hooks/createConsolidateJobsPostHook.js (hunk 2 above)
Comment (antonoberg) [reply packages/backend/src/lib/data/hooks/createConsolidateJobsPostHook.js]: fixed

File: packages/backend/src/lib/data/hooks/createConsolidateJobsPostHook.js (hunk 3 above)
Comment (antonoberg) [reply packages/backend/src/lib/data/hooks/createConsolidateJobsPostHook.js]: fixed

File: package.json (hunk 5 above)
Comment (antonoberg) [reply package.json]: fixed

**Pull request merged**